pkg/codegen/go/gen.go

```diff
@@ -835,6 +876,24 @@ func (pkg *pkgContext) genResource(w io.Writer, r *schema.Resource) error {
 	fmt.Fprintf(w, "\treturn reflect.TypeOf((*%sArgs)(nil)).Elem()\n", camel(name))
 	fmt.Fprintf(w, "}\n\n")
 
+	// Emit the input types.
```

Comment:
Nit: I was initially confused what this was for until I read through the rest of the changes. This is the input type for the resource itself? Maybe clarify the comment?

```suggestion
	// Emit the resource input type.
```

Reply:
Yes, each resource now gets a top level Input/Output type. I'll clarify the comment.